payjoin-cli/src/app.rs

```diff
@@ -209,14 +209,37 @@ impl App {
             .await??;
 
             println!("Sent fallback transaction");
-            match ctx.process_response(&mut response.into_reader()) {
+            let error = match ctx.process_response(&mut response.into_reader()) {
                 Ok(Some(psbt)) => return Ok(psbt),
-                Ok(None) => std::thread::sleep(std::time::Duration::from_secs(5)),
-                Err(re) => {
-                    println!("{}", re);
-                    log::debug!("{:?}", re);
+                Ok(None) => {
+                    std::thread::sleep(std::time::Duration::from_secs(5));
+                    continue;
                 }
-            }
+                Err(error) => error,
```

Comment:
rather than assign `let error = ` and follow this block with logic, I suggest writing be a `handle_response_error(e)` helper which can be called here when `Err(e)` is matched to contain `attempt_version_downgrade` logic and then handle the ultimate error if all else fails.

Reply:
I agree with the concept of abstraction into `attempt_v1` (check spelling) and wonder if it could be abstracted to serve both v1 and v2 features? Does it for sure need to be feature gated? Both v1 and v2 sender attempt v1 sends